plugins/dummy_rand_data/src/hipcheck_transport.rs

```diff
@@ -141,86 +161,86 @@ impl HcTransport {
 			if !msg_chunks.is_empty() {
 				return Err(anyhow!(
 					"received additional messages for id '{}' after QueryComplete status message",
-					id
+					self.id
 				));
 			}
 		}
 		raw.try_into().map(Some)
 	}
 }
+impl Drop for QuerySession {
+	// Remove self from session tracker
+	fn drop(&mut self) {
+		let raw_id = self.id as i32;
+		// @Note - since this will almost always be called from an async function,
+		// we cannot use `blocking_lock()`. this is because the runtime will see
+		// an attempt to block the thread that it is running on and panic. So we
+		// have to do this busy loop style of attempting to acquire the session tracker
+		let mut res_s_handle = self.sessions.try_lock();
+		while res_s_handle.is_err() {
+			res_s_handle = self.sessions.try_lock();
+		}
+		let mut s_handle = res_s_handle.unwrap();
+		eprintln!("Got lock for drop finally");
+		let Some(_channel) = s_handle.remove(&raw_id) else {
+			panic!("Dropping session that does not exist in sessions");
+		};
+	}
+}
```

Comment:
Another option here would be to have a message you can send to the session tracker task telling it that it can drop the session, and then not having the session itself wait for that cleanup, and completing drop here.